File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool cmp(vector<int> v1, vector<int> v2){
    return v1[1] < v2[1];
}
int solution(vector<vector<int>> targets) {
    int answer = 0;
    sort(targets.begin(), targets.end(),cmp);
    
    answer++;
    int point = targets[0][1];
    for(int i = 1; i < targets.size(); i++){
        if(targets[i][0] >= point){
            point = targets[i][1];
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        Arrays.sort(targets, (o1,o2)-> {
            return o1[1] - o2[1];
        });
        
        int curr = -1;
        for(int[] target : targets){
            int st = target[0];
            int ed = target[1];
            if(curr <= st){
                answer++;
                curr = ed;
            }            
        }
        return answer;
    }
}